src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -198,14 +198,16 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
                 LocalDataStoreInstance.LocalDataStore = keyStore;
             }
 
-            services = new DefaultServiceProvider();
+            grainFactory = new GrainFactory();
+            IDictionary<Type,object> singletons = new Dictionary<Type, object> { { typeof(IGrainFactory), grainFactory } };
+            services = new DefaultServiceProvider(singletons);
```

Comment:
The goal is to allow users to plug in their own containers, which I believe is currently the case.

However, for Orleans to use DI, there must be a default implementation.  As I understand it, we intend to use ASP.Net's "Microsoft.Extensions.DependencyInjection" implementation as the default, but it's not been released, so we now use a (laughably minimalistic) placeholder default implementation.